sinatra-contrib/lib/sinatra/content_for.rb

```diff
@@ -171,9 +171,15 @@ def clear_content_for(key)
     #
     # Would pass <tt>1</tt> and <tt>2</tt> to all the blocks registered
     # for <tt>:head</tt>.
-    def yield_content(key, *args)
-      return yield(*args) if block_given? && content_blocks[key.to_sym].empty?
-      content_blocks[key.to_sym].map { |b| capture(*args, &b) }.join
+    def yield_content(key, *args, &block)
+      if block_given? && !content_for?(key)
+        haml? ? capture_haml(*args, &block) : yield(*args)
+      else
+        content = content_blocks[key.to_sym].map { |b| capture(*args, &b) }
+        content.join.tap do |c|
+          @_out_buf << c if block_given? && erb?
+        end
```

Comment:
Why do we need these lines? I guess `@_out_buf` should be reverted after capturing as [this line](https://github.com/sinatra/sinatra/blob/94bd005d933f8ac9394cbe04d331b0b51e93f7bf/sinatra-contrib/lib/sinatra/capture.rb#L102) and it's intended.
If you don't have any concerns about this, please remove L179~181

Reply:
Awesome, your change to the `capture` method eliminates the need to write to `@_out_buf` here!

Reply:
I realized my specs didn't provide the test coverage I expected. I improved them so they cover the scenarios for this issue. If I remove the lines you mentioned, the content will not be written to the template buffer for ERB. We need to write directly to the buffer for ERB because we cannot use the <%= %> expression for block helpers. Padrino handles this in a much more elegant way by using OOP.

Reply:
That makes sense.